tests/unit/clients/python/test_async_client.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+import numpy as np
+
+from jina.flow import Flow
+from jina.excepts import BadClientCallback
+
+
+def validate(x):
+    raise NotImplementedError
+
+
+def test_client_on_error():
+    # This will fail and get stuck
+    # Not sure why this test case was removed
+    f = Flow().add()
+    with pytest.raises(BadClientCallback):
+        with f:
+            f.index_ndarray(np.random.random([5, 4]), output_fn=validate, continue_on_error=False)
+
+    # Recreating the flow instance will make it work.
+    # But reusing the flow upon an error is a good test case
```

Comment:
> # But reusing the flow upon an error is a good test case

But this describes another scenario, which is different than the discussion in Slack.

In this particular test, when you write two tests in a row, you are testing the following case:
- You are testing exception in `callback`, not error in request generator
- The exception breaks the `async stub.Call` 
- Server probably has something hold in the stream
- Restart the client, keep server untouched.
- Now, server stucks

Reply:
https://github.com/grpc/grpc/issues/10974